airflow/contrib/task_runner/cgroup_task_runner.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import os
+import uuid
+
+from cgroupspy import trees
+from cgroupspy.controllers import MemoryController
+
+from airflow.task_runner.base_task_runner import BaseTaskRunner
+
+
+class CgroupTaskRunner(BaseTaskRunner):
```

Comment:
It seems that this class needs root access. That seems excessive. I would expect only cgroup creation handled by root (eg. ``sudo mkdir /sys/fs/cgroup/cpuset/${USER}, sudo chown -R ${USER} /sys/fs/cgroup/cpuset/${USER}``) and then drop privileges. 

Another option could be to have a separate process for cgroup creation but that would be kind of the same as the above.